server/src/main/java/org/apache/cassandra/sidecar/acl/authorization/SidecarPermissions.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.acl.authorization;
+
+/**
+ * Sidecar permissions allowed on specific targets are listed here. Majority of sidecar permissions are represented in
+ * format <action_allowed>:<action_target>.
+ * <p>
+ * Example, with CREATE:SNAPSHOT permission, CREATE action is allowed for SNAPSHOT target. Sample actions are
+ * CREATE, VIEW, UPDATE, DELETE, STREAM, IMPORT, UPLOAD, START, ABORT etc.
+ * <p>
+ * Wildcard permissions are supported with ':' wildcard parts divider and '*' wildcard token to match parts:
+ * <p>
+ * - *:SNAPSHOT allows CREATE:SNAPSHOT, VIEW:SNAPSHOT and DELETE:SNAPSHOT.
+ * - CREATE:* allows CREATE action on all possible targets.
+ * - *:* allows all possible permissions for specified resource
+ */
+public class SidecarPermissions
+{
+    // cassandra cluster related permissions
+    public static final Permission VIEW_CLUSTER = new WildcardPermission("VIEW:CLUSTER");
+
+    // SSTable related permissions
+    public static final Permission UPLOAD_SSTABLE = new WildcardPermission("UPLOAD:SSTABLE");
+    public static final Permission IMPORT_SSTABLE = new WildcardPermission("IMPORT:SSTABLE");
+    public static final Permission STREAM_SSTABLE = new WildcardPermission("STREAM:SSTABLE");
```

Comment:
What about using `READ` to replace both `VIEW` and `STREAM`? Do we need to distinguish between those 2 read operations?

Reply:
As of now I don't see any need to distinguish them

Reply:
Should we have `READ` and `STREAM`. We can use read for listing snapshots, reading topology, listing restore job in place of `VIEW` to align with http methods. `STREAM` feels like a separate action, hence distinction?

Reply:
One for use case for separating `READ` and `STREAM` . We have to allow permission for listing cdc segments and streaming them, currently we have `READ:CDC` and `STREAM:CDC` separately.